### PR TITLE
Move to CI workflows that allow but don't require secrets

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

**Review by maraino — CHANGES_REQUESTED** [review]

File: .github/workflows/goCI.yml [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+    with:
+      cgo: ${{ inputs.cgo }}
+    secrets: inherit
 
   build:
-    uses: smallstep/workflows/.github/workflows/goBuild.yml@main
+    uses: smallstep/workflows/.github/workflows/goBuild2.yml@main
+    with:
+      cgo: ${{ inputs.cgo }}
+    secrets: inherit
```
[the remark this answers is not in the document]
Comment (dopey) [reply .github/workflows/goCI.yml:32]: ooooooops

File: .github/workflows/goBuild.yml [hunk 2]
```diff
@@ -8,15 +17,37 @@ jobs:
...
     steps:
-      - name: Install Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: ${{ matrix.go }}
-          check-latest: true
       - name: Install libpcsclite-dev # Some dependencies require this package
+        if: ${{ inputs.cgo }}
         run: |
           sudo apt-get install libpcsclite-dev
```
Comment (maraino) [inline .github/workflows/goBuild.yml:23]: It will make sense to add the list of dependencies as input.

File: .github/workflows/goLint.yml [hunk 3]
```diff
@@ -1,20 +1,51 @@
...
+        required: false
 
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
+      - name: Install libpcsclite-dev # Some dependencies require this package
+        if: ${{ inputs.cgo }}
+        run: |
+          sudo apt-get install libpcsclite-dev
```
Comment (maraino) [inline .github/workflows/goLint.yml:20]: Get the list of dependencies as input.

File: .github/workflows/goTest.yml [hunk 4]
```diff
@@ -8,6 +19,33 @@ jobs:
       matrix:
         go: ['1.18', '1.19']
     steps:
+      - name: Install libpcsclite-dev # Some dependencies require this package
+        if: ${{ inputs.cgo }}
+        run: |
+          sudo apt-get install libpcsclite-dev
```
Comment (maraino) [inline .github/workflows/goTest.yml:25]: Ditto

**Review by maraino — CHANGES_REQUESTED** [review]

File: .github/workflows/goBuild.yml [hunk 5]
```diff
@@ -8,15 +17,37 @@ jobs:
       matrix:
         go: ['1.18', '1.19']
```
Comment (maraino) [inline .github/workflows/goBuild.yml:18]: Do we need the matrix here?

**Review by maraino — APPROVED** [review]
LGTM, but I added a comment.

File: .github/workflows/goBuild.yml (hunk 5 above)
Comment (dopey) [reply .github/workflows/goBuild.yml:18]: We don't, but we'll make this parameterizable in a future PR.

**Pull request merged**